### PR TITLE
test: use test-launch.sh to run the tests

### PR DESCRIPTION
**Pull request by goneri**
Use one single script called `./tools/test-launch.sh` to trigger all the different tests:

- `COVERAGE=1` or `-c` to enable the code coverage
- `COVE_VERSION` to set the vscode-extension-tester parameter (e.g `max` or `min`)
- `MOCK_LIGHTSPEED_API` (`1` or ``): set it to `1` to dynamically start the mocked server and set `TEST_LIGHTSPEED_URL`
- `TEST_TYPE`: `e2e` or `ui` (default)

The goal is to reduce the amount of reentrant `yarn` calls and provide a more linar execution flow.
Bonus points:
- we don't need `start-server-and-test` anymore
- we avoid a lot of complexity with how we deal with `TEST_LIGHTSPEED_URL` on MacOS.

### CODE REVIEW

**Review by ssbarnea — COMMENTED** [review]
I love the direction. My only question is why the yarn.lock file is modified when you did not modify package.json in any way?

File: package.json [hunk 1]
```diff
@@ -917,7 +917,6 @@
     "marked": "^14.1.3",
     "minimatch": "^10.0.1",
     "semver": "^7.6.3",
-    "start-server-and-test": "^2.0.8",
```
Comment (goneri) [inline package.json:920]: @ssbarnea The change is here ^

**Comment by goneri** [comment]
> I love the direction. My only question is why the yarn.lock file is modified when you did not modify package.json in any way?

https://github.com/ansible/vscode-ansible/pull/1649/files#r1834505006

**Review by ssbarnea — APPROVED** [review]

**Review by ssbarnea — APPROVED** [review]

**Review by TamiTakamiya — COMMENTED** [review]
Thank you @goneri!  This was long due change that I should have made months ago...

**Pull request merged**